mt_sat/mt_sat_wrapper.m

```diff
@@ -78,12 +78,33 @@ function mt_sat_wrapper(mtw_nii,pdw_nii,t1w_nii,mtw_jsn,pdw_jsn,t1w_jsn,varargin
 % This env var will be consumed by qMRLab
 setenv('ISNEXTFLOW','1');
 
-if nargin >6
-if any(cellfun(@isequal,varargin,repmat({'qmrlab_path'},size(varargin))))
-    idx = find(cellfun(@isequal,varargin,repmat({'qmrlab_path'},size(varargin)))==1);
-    qMRdir = varargin{idx+1};
+p = inputParser();
+
+%Input parameters conditions
+validNii = @(x) exist(x,'file') && strcmp(x(end-5:end),'nii.gz');
+validJsn = @(x) exist(x,'file') && strcmp(x(end-3:end),'json');
+validB1factor = @(x) isnumeric(x) && (x > 0 && x <= 1);
+
+%Add REQUIRED Parameteres
+addRequired(p,'mtw_nii',validNii);
+addRequired(p,'pdw_nii',validNii);
+addRequired(p,'t1w_nii',validNii);
+addRequired(p,'mtw_jsn',validJsn);
+addRequired(p,'pdw_jsn',validJsn);
+addRequired(p,'t1w_jsn',validJsn);
+
+%Add OPTIONAL Parameteres
+addParameter(p,'mask',validNii);
+addParameter(p,'b1map',validNii);
+addParameter(p,'b1factor',validB1factor);
+addParameter(p,'qmrlab_path',@ischar);
+addParameter(p,'sid',@ischar);
+
+parse(p,mtw_nii,pdw_nii,t1w_nii,mtw_jsn,pdw_jsn,t1w_jsn,varargin{:});
+
+if any(cellfun(@(x) isequal(x,'qmrlab_path'),varargin))
```

Comment:
As in here.

```diff
- if any(cellfun(@(x) isequal(x,'qmrlab_path'),varargin))
+ isempty(p.Results.qmrlab_path)
```

Reply:
Requires respective changes made above.

Reply:
@agahkarakuzu in this [commit](https://github.com/qMRLab/qMRWrappers/pull/2/commits/2283c0d1cb8f1165da876f8b8a0b1f1ac11a7489) I checked the optional arguments using `isempty` but I get those arguments using something like this: 
```
if ~isempty(p.Results.qmrlab_path)
      qMRdir = p.Results.qmrlab_path
end
```

Reply:
@jvelazquez-reyes this works, you can collapse these if statements to a single line and remove `if nargin>6` (L135) condition as it is now redundant.

Reply:
@agahkarakuzu I removed `if nargin>6` and collapsed the if statements. Unless you indicate other details, I will resolve this comment.